contracts/factories/DelegatedManagerFactory.sol

```diff
@@ -220,48 +220,62 @@ contract DelegatedManagerFactory {
+     * @param  _extensions              List of addresses of extensions which need to be initialized
      * @param  _initializeBytecode      List of bytecode encoded calls to relevant target's initialize function
      */
     function initialize(
         ISetToken _setToken,
         uint256 _ownerFeeSplit,
         address _ownerFeeRecipient,
-        address[] memory _initializeTargets,
+        address[] memory _extensions,
         bytes[] memory _initializeBytecode
     )
         external
     {
         require(initializeState[_setToken].isPending, "Manager must be awaiting initialization");
         require(msg.sender == initializeState[_setToken].deployer, "Only deployer can initialize manager");
-        _initializeTargets.validatePairsWithArray(_initializeBytecode);
-
-        for (uint256 i = 0; i < _initializeTargets.length; i++) {
-            address target = _initializeTargets[i];
-            require(!controller.isSet(target), "Target must not be SetToken");
-
-            // Because we validate uniqueness of _initializeTargets only one transaction can be sent to each module or extension during this
-            // transaction. Due to this no modules/extension can be used for any SetToken transactions other than initializing these contracts
-            target.functionCallWithValue(_initializeBytecode[i], 0);
-        }
+        _extensions.validatePairsWithArray(_initializeBytecode);
 
         IDelegatedManager manager = initializeState[_setToken].manager;
-        manager.updateOwnerFeeSplit(_ownerFeeSplit);
-        manager.updateOwnerFeeRecipient(_ownerFeeRecipient);
 
         // If the SetToken was factory-deployed & factory is its current `manager`, transfer
         // managership to the new DelegatedManager
         if (_setToken.manager() == address(this)) {
             _setToken.setManager(address(manager));
         }
 
-        manager.transferOwnership(initializeState[_setToken].owner);
-        manager.setMethodologist(initializeState[_setToken].methodologist);
+        for (uint256 i = 0; i < _extensions.length; i++) {
+            address extension = _extensions[i];
+            require(managerCore.isExtension(extension), "Target must be ManagerCore-enabled Extension");
+
+            bytes memory initializeBytecode = _initializeBytecode[i];
+
+            // Each input initializeBytecode is a varible length bytes array which consists of a 32 byte prefix for the
+            // length parameter, a 4 byte function selector, a 32 byte DelegatedManager address, and any additional parameters
+            // as shown below:
+            // [32 bytes - length parameter, 4 bytes - function selector, 32 bytes - DelegatedManager address, additional parameters]
+            // It is required that the input DelegatedManager address is the DelegatedManager address corresponding to the caller
+            address inputManager;
```

Comment:
Sorry to do this but jsut to clean things up a bit can we put the initialization logic into its own internal function? Can pass in the `_extensions`, `_initializeBytecode`, and `manager`

Reply:
No problemo - looks much cleaner